earth_moving/clutter/engine2.py

```diff
@@ -0,0 +1,346 @@
+# this file contains the PyBulletEnvironment class
+# the general capabilities are opening the environment, closing the environment, and general control functions
+
+# general imports
+import pybullet as p
+import pybullet_data
+import time as t
+import numpy as np
+import math
+
+# class definition
+class PyBulletEnvironment:
+    
+    # constructor
+    def __init__(self, gui=True, gravity=(0, 0, -10)):
+        """
+        Initialize the PyBullet environment.
+
+        :param gui: Boolean, if True the environment will be opened with a GUI.
+        :param gravity: Tuple, the gravity vector.
+        """
+        
+        # gui
+        self.gui = gui
+        self.physicsClient = None 
+        
+        # control accuracy threshold        
+        self.control_threshold = 1e-2 
+        
+        # iter limit
+        self.iter_limit = 5e2
+        
+        # ID list 
+        self.ID = []
```

Comment:
I wonder about that...
(1) Why not make it a dictionary in which we map a semantic name ("plane", "robot", "pebble1", "pebble17", etc.) to its ID?
(2) In pybullet you have various types of IDs, and it's always a bit confusion. Let's be specific: if these are "body unique IDs" (that's how they're called?), I suggest naming them `uids` or `body_ids`.

Reply:
yes they are body unique IDs, all of them. Is there a way to "bind" the dictionary name to the ID? Like a symlink in linux.